r/configure

```diff
@@ -255,17 +254,19 @@ if [ $? -eq 0 ] || [ "$UNAME" = "Darwin" ]; then
   echo "PKG_CFLAGS=$PKG_CFLAGS"
   echo "PKG_LIBS=$PKG_LIBS"
 else
-  if [ "$UNAME" = "Darwin" ]; then
-    # Just for debugging: is this possible?
-    echo "Test to load header failed. Command:"
-    echo "$TEST_CMD"
-  fi
   echo "------------------------- NOTE ---------------------------"
   echo "See https://arrow.apache.org/docs/r/articles/install.html"
   echo "for help installing Arrow C++ libraries"
   echo "---------------------------------------------------------"
```

Comment:
```suggestion
  echo "------------------------- NOTE ---------------------------"
  echo "There was an issue building or finding the Arrow C++ libraries."
  echo "See https://arrow.apache.org/docs/r/articles/install.html"
  echo "for help installing Arrow C++ libraries"
  echo "---------------------------------------------------------"
```

I added a bit to make it a little more clear/obvious what happened there. We have the URL which explains it, but I think stating definitively that there was a problem building/finding is good/makes this more understandable. Open to other wordings of course.